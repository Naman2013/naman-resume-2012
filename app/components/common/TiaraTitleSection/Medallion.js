import React from 'react';
import PropTypes from 'prop-types';
import { golda, blue_tile_feat } from '../../../styles/variables/colors_tiles_v4';

function calculateIconSize(dimension) {
  const iconToDimensionRatio = 0.5;
  return (dimension * iconToDimensionRatio);
}

const Medallion = ({ iconURL, dimension, theme }) => (
  <div
    style={theme}
  >
    <img
      alt=""
      width={calculateIconSize(dimension)}
      height={calculateIconSize(dimension)}
      src={iconURL}
    />
    <style jsx>
      {`
        div {
          position: relative;
          width: ${dimension}px;
          height: ${dimension}px;
          border-radius: 50%;
          border: 5px solid ${golda};
          background: url(${blue_tile_feat});
          margin: 0;
          padding: 0;
        }

        img {
          position: absolute;
          top: 50%;
          left: 50%;
          transform: translate(-50%, -50%);
        }
      `}
    </style>
  </div>
);

Medallion.propTypes = {
  iconURL: PropTypes.string.isRequired,
  theme: PropTypes.shape({}),
  dimension: PropTypes.number,
};

Medallion.defaultProps = {
  theme: {},
  dimension: 50,
};

export default Medallion;
