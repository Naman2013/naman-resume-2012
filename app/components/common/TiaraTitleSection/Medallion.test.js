import React from 'react';
import { shallow } from 'enzyme';
import Medallion from './Medallion';

const setup = customProps => {
  const props = Object.assign({ iconURL: '' }, customProps);
  const shallowWrapper = shallow(<Medallion {...props} />);
  return {
    shallowWrapper,
  };
};

describe('Medallion', () => {
  it('should render correctly', () => {
    expect(setup().shallowWrapper).toMatchSnapshot();
  });
});
