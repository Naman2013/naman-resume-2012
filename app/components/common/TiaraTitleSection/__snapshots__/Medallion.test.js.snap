// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medallion should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Medallion
    dimension={50}
    iconURL=""
    theme={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className="jsx-1285550702"
          height={25}
          src=""
          width={25}
        />,
        <JSXStyle
          css={
            Array [
              "div.__jsx-style-dynamic-selector{position:relative;width:50px;height:50px;border-radius:50%;border:6px solid #e2b979;background:url(https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png);margin:0;padding:0;}",
              "img.__jsx-style-dynamic-selector{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
            ]
          }
          dynamic={
            Array [
              50,
              50,
              "#e2b979",
              "https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png",
            ]
          }
          styleId="4212983649"
        />,
      ],
      "className": "jsx-1285550702",
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "jsx-1285550702",
          "height": 25,
          "src": "",
          "width": 25,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "div.__jsx-style-dynamic-selector{position:relative;width:50px;height:50px;border-radius:50%;border:6px solid #e2b979;background:url(https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png);margin:0;padding:0;}",
            "img.__jsx-style-dynamic-selector{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
          ],
          "dynamic": Array [
            50,
            50,
            "#e2b979",
            "https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png",
          ],
          "styleId": "4212983649",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="jsx-1285550702"
            height={25}
            src=""
            width={25}
          />,
          <JSXStyle
            css={
              Array [
                "div.__jsx-style-dynamic-selector{position:relative;width:50px;height:50px;border-radius:50%;border:6px solid #e2b979;background:url(https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png);margin:0;padding:0;}",
                "img.__jsx-style-dynamic-selector{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
              ]
            }
            dynamic={
              Array [
                50,
                50,
                "#e2b979",
                "https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png",
              ]
            }
            styleId="4212983649"
          />,
        ],
        "className": "jsx-1285550702",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "jsx-1285550702",
            "height": 25,
            "src": "",
            "width": 25,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "div.__jsx-style-dynamic-selector{position:relative;width:50px;height:50px;border-radius:50%;border:6px solid #e2b979;background:url(https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png);margin:0;padding:0;}",
              "img.__jsx-style-dynamic-selector{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
            ],
            "dynamic": Array [
              50,
              50,
              "#e2b979",
              "https://vega.slooh.com/assets/v4/common/Dark_Pattern_Featured_LargeCanvas.png",
            ],
            "styleId": "4212983649",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
